File: Example/Example/IAMDelegate.m
//
//  IAMDelegate.m
//  Example
//
//  Created by ShiCheng Lu on 7/23/25.
//  Copyright © 2025 Radar Labs, Inc. All rights reserved.
//

#import <Foundation/Foundation.h>

#import "IAMDelegate.h"

@implementation MyObjC_IAMDelegate

- (void)onInAppMessageButtonClicked:(RadarInAppMessage *)message {
    NSLog(@"onIAMPositiveAction ObjC");
}

@end

File: Example/Example/IAMDelegate.swift
//
//  IAMDelegate.swift
//  Example
//
//  Created by ShiCheng Lu on 7/22/25.
//  Copyright © 2025 Radar Labs, Inc. All rights reserved.
//

import Foundation
import RadarSDK
import SwiftUICore


class MyIAMDelegate: RadarInAppMessageDelegate {
    override func onInAppMessageButtonClicked(_ message: RadarInAppMessage) {
        if let url = message.button?.url {
            UIApplication.shared.open(URL(string: url)!)
        }
        print("custom on IAM positive action")
        // possibly log conversion
    }
}

